cookbooks/aws-parallelcluster-environment/files/cloudwatch/write_cloudwatch_agent_json.py

```diff
@@ -45,15 +56,19 @@ def gethostname():
     return socket.gethostname().split(".")[0]
 
 
-def write_config(config):
-    """Write config to AWS_CLOUDWATCH_CFG_PATH."""
-    with open(AWS_CLOUDWATCH_CFG_PATH, "w+", encoding="utf-8") as output_config_file:
+def write_config(config, output_file):
+    """Write config to output_file."""
+    with open(output_file, "w+", encoding="utf-8") as output_config_file:
         json.dump(config, output_config_file, indent=4)
 
 
-def add_log_group_name_params(log_group_name, configs):
+def add_log_group_name_params(default_log_group_name, log_group_name_map, configs):
     """Add a "log_group_name": log_group_name to every config."""
     for config in configs:
+        log_group_name = default_log_group_name
+        group_key = config.get("log_group_key")
```

Comment:
I think you need to provide a default here: `.get("log_group_key", None)` otherwise this call will throw an error if "log_group_key" is not present in `config`.